contracts/cctp/SynapseCCTP.sol

```diff
@@ -0,0 +1,123 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.17;
+
+import {SynapseCCTPEvents} from "./events/SynapseCCTPEvents.sol";
+import {IMessageTransmitter} from "./interfaces/IMessageTransmitter.sol";
+import {ISynapseCCTP} from "./interfaces/ISynapseCCTP.sol";
+import {ITokenMessenger} from "./interfaces/ITokenMessenger.sol";
+import {RequestLib} from "./libs/Request.sol";
+
+contract SynapseCCTP is SynapseCCTPEvents, ISynapseCCTP {
+    // TODO: add setters for these (or make them immutable)
+    uint32 public localDomain;
+    IMessageTransmitter public messageTransmitter;
+    ITokenMessenger public tokenMessenger;
+    mapping(uint32 => bytes32) public remoteSynapseCCTP;
+
+    /// @inheritdoc ISynapseCCTP
+    function sendCircleToken(
+        address recipient,
+        uint32 destinationDomain,
+        address burnToken,
+        uint256 amount,
+        uint32 requestVersion,
+        bytes memory swapParams
```

Comment:
Might want to consider checking swap params are valid, if exist, on origin chain before bridging even with transfer fallback

Reply:
`RequestLib.formatRequest()` will revert if the params are not properly formatted (basically if the length does not match the correct one). Other than that, what kind of checks do we want to do here?

I guess we could decode and check that values are passing the sanity checks (pool address is not zero, tokenFrom index differs from tokenTo index), but anything beyond that is hard, as the pool is on the dest chain and we can't really check the params much.

Also worth noting that even for a perfectly invalid swapParams payload (a bunch encoded zeroes), the swap on dest chain will just fail, and recipient **will receive the bridged Circle token instead** (will add this test to #239 later), so having a `swapParams` with some nonsense values won't break things.

Reply:
@mikeyrf 
- The above test was added in a25fbff.
- Fix for making sure invalid `swapParams` inputs don't lead to tx being reverts on dest chain was added in 251f1f2.